### PR TITLE
Document constants in Transform/Transform2D

### PR DESCRIPTION
**Pull request by marcotc**
Add documentation for `Transform` and `Transform2D` class constants.

### CODE REVIEW

File: doc/classes/Transform2D.xml [hunk 1]
```diff
@@ -172,10 +172,13 @@
 	</members>
 	<constants>
 		<constant name="IDENTITY" value="Transform2D( 1, 0, 0, 1, 0, 0 )">
+			[Transform2D] with no translation, rotation or scaling applied. When applied to other data structures, [constant IDENTITY] performs no transformation.
 		</constant>
 		<constant name="FLIP_X" value="Transform2D( -1, 0, 0, 1, 0, 0 )">
+			[Transform2D] with vertical mirroring applied.
```
Comment (Calinou) [inline doc/classes/Transform2D.xml]: I suppose this is correct (I haven't played with transforms much), but are you sure it's vertical mirroring here?

**Review by Calinou — APPROVED** [review]
Looks good to me :slightly_smiling_face:

File: doc/classes/Transform2D.xml (hunk 1 above)
Comment (marcotc) [reply doc/classes/Transform2D.xml]: I was pretty sure until you mentioned it, let me run a quick test locally.

Comment (marcotc) [reply doc/classes/Transform2D.xml]: Ok `FLIP_X` mirrors left-right, `FLIP_Y` mirrors up-down.

I think the language is ambiguous. I'll change it up.

Comment (marcotc) [reply doc/classes/Transform2D.xml]: I think it's more clear, albeit more technical now.

File: doc/classes/Transform2D.xml [hunk 2]
```diff
@@ -172,10 +172,13 @@
 	</members>
 	<constants>
 		<constant name="IDENTITY" value="Transform2D( 1, 0, 0, 1, 0, 0 )">
+			[Transform2D] with no translation, rotation or scaling applied. When applied to other data structures, [constant IDENTITY] performs no transformation.
 		</constant>
 		<constant name="FLIP_X" value="Transform2D( -1, 0, 0, 1, 0, 0 )">
+			[Transform2D] with mirroring applied parallel the X axis.
```
Comment (Calinou) [inline doc/classes/Transform2D.xml]:
```suggestion
			[Transform2D] with mirroring applied parallel to the X axis.
```

File: doc/classes/Transform2D.xml [hunk 3]
```diff
@@ -172,10 +172,13 @@
...
 		</constant>
 		<constant name="FLIP_Y" value="Transform2D( 1, 0, 0, -1, 0, 0 )">
+			[Transform2D] with mirroring applied parallel the Y axis.
```
Comment (Calinou) [inline doc/classes/Transform2D.xml]:
```suggestion
			[Transform2D] with mirroring applied parallel to the Y axis.
```

**Review by Calinou — CHANGES_REQUESTED** [review]
There's a missing word, but it's fine to me otherwise :slightly_smiling_face:

File: doc/classes/Transform2D.xml (hunk 3 above)
Comment (marcotc) [reply doc/classes/Transform2D.xml]: Addressed on both cases.

**Review by Calinou — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by marcotc** [comment]
I just wanted to say that this was the fastest turnaround of any open source project I've ever contributed to.
You guys are doing an amazing job here!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**